service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -648,9 +648,10 @@ public boolean updateResource(
   public void createControlledResource(ControlledResource controlledResource)
       throws DuplicateResourceException {
 
-    if (!cloudContextExists(
```

Comment:
nit: can you add a resourcedao test that create a controlled flexible resource?

Reply:
added.